internal/pod/kmodloader.go

```diff
@@ -0,0 +1,551 @@
+package pod
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/config"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/meta"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/worker"
+	"github.com/mitchellh/hashstructure/v2"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubectl/pkg/cmd/util/podcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+)
+
+//go:generate mockgen -source=kmodloader.go -package=pod -destination=mock_kmodloader.go KmodLoader
+
+type KmodLoader interface {
```

Comment:
"KmodLoader" is not a good name for this interface. I was looking at the main.go and got really confused when i encountered that interface's initialization function. I think something like "WorkerPodManager/Handler" would be more appropriate and clear.